### ISSUE TITLE
[v1 QUESTION]: Hi i used chemprop to run a simulation for dataset size 1% of 1M

### ISSUE THREAD

**Issue by MAK0077** [chemprop/chemprop#1334]
**What are you trying to do?**
Please tell us what you're trying to do with Chemprop, providing as much detail as possible
I am using chemprop just to see why the model misses good compounds but i dont know how can i do this and this is my thesis topic i want to end my thesis until end of May 2026 and i am so stressful now.Right now i tried to plot the tsne of  recalled compounds, missed compounds, traning set and other compounds 
**Previous attempts**
If possible, provide some examples of what you've already tried and what the output was.
my supervisor said to me plot the recalled compounds with good docking score recalled compounds,medium and bad and same for missed compounds with different colours and if the missed compounds have a good docmking score then its problem why model missed them and then find why model is missing them but i dont know how can i do this because i never used these softwares before and i dont know how to do coding and write python scripts and its getting so difficult and stressful for me day by day

**Screenshots**

**Comment by KnathanM** [comment]
Hi @MAK0077, sorry to hear research is stressful. Please do reach out to your research institution's support network if you feel you could use talking with someone about how to manage that stress. 

It looks like you are using chemprop v1 and want to make a TSNE plot of the test set and training set to see if the molecules that the model performed poorly on are different from what was in the training set. 

The first step is to get a vector representation for each molecule. In chemprop v1, I believe this is described [here](https://chemprop.readthedocs.io/en/v1.7.1/models.html#chemprop.models.model.MoleculeModel.fingerprint). For chemprop v2 the instructions are [here](https://chemprop.readthedocs.io/en/latest/mpnn_fingerprints.html). 

Then you can reduce the dimension of the vectors using TSNE or PCA, and plot all the points in 2D, with different colors for training points, good test predictions, and bad test predictions. An LLM like ChatGPT should be helpful with this. We also have a small example of PCA [here](https://chemprop.readthedocs.io/en/latest/mpnn_fingerprints.html) that you could adapt. 

Then hopefully the plot shows that poor predictions are for molecules different than the training data. Probably it won't be exact as ML research usually isn't, but it may be good enough for your supervisor. Good luck!

As a side note, you can also convert a chemprop v1 model to v2 using the instructions [here](https://chemprop.readthedocs.io/en/latest/tutorial/cli/convert.html).
